### PR TITLE
Sieze the means of production

### PR DESCRIPTION
**Pull request by sshirokov**
This takes the python driver actually controlling the printer, and puts us in control of its interface without having to rely on getting the changes inside s3g or a fork of it.

The driver as it stands is a copy of the example script that was used before, but with added traces to output the progress of the job.

Some more changes need to be made to clean up s3g's modules, then this script's loader can become much less hideous.

The operation of the raw python script is wrapped by a shell script that activates the s3g virtual environment to move that logic out of the makefile.

/cc @skalnik

### CODE REVIEW

**Comment by skalnik** [comment]
Looks good, but I'd like to test before merging :+1:

File: bin/print_gcode.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+from datetime import datetime
+import os
+import sys
+lib_path = os.path.abspath(os.path.dirname(__file__) + '/../vendor/s3g')
+sys.path.append(lib_path)
```
Comment (skalnik) [inline bin/print_gcode.py:6]: Funsies

Comment (sshirokov) [inline bin/print_gcode.py:6]: That's the part that I'd rather fix by unfucking the `s3g` python package correctly.

File: bin/print_gcode.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        parser.execute_line(line)
+
+print "==%s==> Starting gcode stream" % datetime.now()
+with open(options.filename) as f:
+    lines = list(f)
+    num_lines = len(lines)
+    for n, line in enumerate(lines):
+        parser.execute_line(line)
+        percent = (n / num_lines) * 100
+        print "==%s==> Sent %d/%d [%d%%]" % (datetime.now(), n, num_lines, percent)
```
Comment (skalnik) [inline bin/print_gcode.py]: We should only print the percentage every 5/10%, not every line of GCode we have.

Comment (sshirokov) [inline bin/print_gcode.py]: 3aec93d -- % 5

**Comment by skalnik** [comment]
IT VERKS!

**Pull request merged**